Sources/StreamChat/Utils/KeyedDecodingContainer+Array.swift

```diff
@@ -6,25 +6,83 @@ import Foundation
 
 private struct ElementWrapper<T: Decodable>: Decodable {
     let value: T?
+    var rawJSON: [String: RawJSON]?
     var error: Error?
     init(from decoder: Decoder) throws {
         do {
             value = try T(from: decoder)
         } catch {
             value = nil
             self.error = error
+            // We decode the payload as RawJSON to be able to display the payload in case of error
+            rawJSON = try? [String: RawJSON](from: decoder)
         }
     }
 }
 
 extension KeyedDecodingContainer {
     func decodeArrayIgnoringFailures<T: Decodable>(_ type: [T].Type, forKey key: KeyedDecodingContainer<K>.Key) throws -> [T] {
-        let wrapper = try decode([ElementWrapper<T>].self, forKey: key)
-        let errors = wrapper.compactMap(\.error)
-        if !errors.isEmpty {
-            let errorsDescription = errors.map(String.init(describing:))
-            log.error("Failed decoding elements from array: \(errorsDescription)", subsystems: .httpRequests)
+        let wrappers = try decode([ElementWrapper<T>].self, forKey: key)
+        for wrapper in wrappers where wrapper.error != nil {
+            let rawJSONPrettyPrinted = (try? JSONEncoder.default.encode(wrapper.rawJSON))?.debugPrettyPrintedJSON
```

Comment:
I am worried about the double decode/encode here. Why are you decoding `rawJSON` in the first place if you are going to encode it here?

Reply:
RawJSON conversion is only done when a decoding fails, so no overhead for the good path. Worst case, every element in an array where we use `decodeArrayIgnoringFailures` will fail, so we'll decode them into RawJSON, which will double the decoding time for the said array.

There's no way (that I could find) to get the Raw JSON data from decoder, that's why I opted for the RawJSON decoding path, to be able to print the erroring payload